### PR TITLE
New poster view release date

### LINKED ISSUE #5656: Addition of Release dates on Poster view mode
**Is your feature request related to a problem? Please describe.**
Feature QOL Request

**Describe the solution you'd like**
An ability on Poster view, to toggle release dates/digital/physical release dates.

**Describe alternatives you've considered**
I tried list view, but they have no banners for the movies.

The following is where I mean the info to show
https://i.imgur.com/TAEy7Cj.png

### PR DESCRIPTION
**Pull request by RobinDadswell**
#### Database Migration
NO

#### Description
Release date as a calculated value as an option on the Poster View - uses dates in this order

1. Earliest of physical/digital release (if both specified)
2. digital/physical release (if only one specified)
3. cinema release date

#### Screenshot (if UI related)
image

#### Todos
- [x] Translation Keys

#### Issues Fixed or Closed by this PR

* Fixes #5656

### CODE REVIEW

File: frontend/src/Movie/Index/Posters/Options/MovieIndexPosterOptionsModalContent.js [hunk 1]
```diff
@@ -174,6 +181,18 @@ class MovieIndexPosterOptionsModalContent extends Component {
               />
             </FormGroup>
 
+            <FormGroup>
+              <FormLabel>{translate('ShowReleaseDate')}</FormLabel>
```
Comment (Qstick) [inline frontend/src/Movie/Index/Posters/Options/MovieIndexPosterOptionsModalContent.js:185]: Add this one to en.json as well

**Comment by Qstick** [comment]
One other note, even if Show Quality Profile is off we still show the value on a sort by QualityProfile.. We used to show the individual release dates on sort, we should likely do the same name even with the prop off.

File: frontend/src/Movie/Index/Posters/MovieIndexPoster.js [hunk 2]
```diff
@@ -253,9 +275,25 @@ class MovieIndexPoster extends Component {
...
+            <div className={styles.title}>
+              {getRelativeDate(
+                releaseDate,
+                shortDateFormat,
+                showRelativeDates,
+                {
+                  timeFormat,
+                  timeForToday: false
+                }
+              )}
```
Comment (austinwbest) [inline frontend/src/Movie/Index/Posters/MovieIndexPoster.js:289]: `{getRelativeDate(releaseDate, shortDateFormat, showRelativeDates, { timeFormat, timeForToday: false })}` for consistency with other usages

File: frontend/src/Movie/Index/Posters/MovieIndexPoster.js [hunk 3]
```diff
@@ -127,6 +132,23 @@ class MovieIndexPoster extends Component {
...
+          releaseDate = digitalRelease;
+        }
+      } else if (physicalRelease && !digitalRelease) {
+        releaseDate = physicalRelease;
+      } else if (digitalRelease && !physicalRelease) {
+        releaseDate = digitalRelease;
+      } else {
+        releaseDate = inCinemas;
+      }
+    }
```
Comment (austinwbest) [inline frontend/src/Movie/Index/Posters/MovieIndexPoster.js:150]: `    let releaseDate = '';
    if (showReleaseDate) {
	  releaseDate = inCinemas;
      if (physicalRelease && digitalRelease) {
	    releaseDate = (physicalRelease < digitalRelease) ? physicalRelease : digitalRelease;
      } else if (physicalRelease) {
        releaseDate = physicalRelease;
      } else if (digitalRelease) {
        releaseDate = digitalRelease;
	  }
    }`

I think that (or close to it, i didn't test it) would simplify it a little?

Comment (RobinDadswell) [reply frontend/src/Movie/Index/Posters/MovieIndexPoster.js:150]: That ternary operator was exactly the line I couldn't think of when I was writing it! I kept saying to myself that there was a different way to do it! I have updated that and the commit is incoming

File: frontend/src/Movie/Index/Posters/MovieIndexPoster.js (hunk 2 above)
Comment (RobinDadswell) [reply frontend/src/Movie/Index/Posters/MovieIndexPoster.js:289]: I got this format from MovieIndexPosterInfo.js (used for sorting) - I am happy to change it but there seems to be a mixed consistency on this one already

**Comment by RobinDadswell** [comment]
> One other note, even if Show Quality Profile is off we still show the value on a sort by QualityProfile.. We used to show the individual release dates on sort, we should likely do the same name even with the prop off.

@Qstick  if I am understanding this right you want to be able to sort by this calculated release date value yeah?

**Comment by Qstick** [comment]
> > One other note, even if Show Quality Profile is off we still show the value on a sort by QualityProfile.. We used to show the individual release dates on sort, we should likely do the same name even with the prop off.
> 
> @Qstick if I am understanding this right you want to be able to sort by this calculated release date value yeah?

Not a bad idea, but no I was talking about the current behavior where the date variant is shown below the poster if the sort is set to that date type (see below from current dev). Make sure we maintain this functionality even if Show Release Dates is off. Like we do for Quality Profiles

image

**Comment by RobinDadswell** [comment]
That is currently broken with this other option so I will look at why and fix it :)

**Comment by RobinDadswell** [comment]
Updated to have In Cinemas or Digital/Released next to relevant dates
image

Ready for review again

**Comment by austinwbest** [comment]
There are icons used for that, on both the calendar & details pages. Takes less room too. Just a thought to keep consistent :)

**Comment by RobinDadswell** [comment]
They aren't used with the dates used for sorting on the posters which is why it's not using them

**Comment by austinwbest** [comment]
That's because there was only one previously? They where not on the calendar either when it was only cinema, I added them when the other dates where added :)

**Comment by RobinDadswell** [comment]
I was going for consistency with these this https://github.com/Radarr/Radarr/blob/f0bb614360bcc2df39ccea69b2dbc975e389a1a2/frontend/src/Movie/Index/Posters/MovieIndexPosterInfo.js#L73 (the next couple of blocks deal with Digital and Physical as well) otherwise they should also be amended to use the icons not the wording for consistency

**Comment by RobinDadswell** [comment]
After some discussion with @austinwbest  this is now set to look like 
image
The sort when they aren't selected is also set to add the icons; 
@Qstick @austinwbest this is ready for a review and hopefully can be pulled in

**Pull request merged**